text/0000-api-version-compat.md

```diff
@@ -0,0 +1,127 @@
+If the buildpack API declared in the buildpack.toml of an optional buildpack is not supported by a given lifecycle, the lifecycle will warn, omit this buildpack from the group, and continue running.
+
+## Deprecations
+If a feature or the usage of a feature (flag names, key names, etc.) is deprecated in a given API in the spec, and a platform or buildpack uses the deprecated feature with that API, the lifecycle will print a deprecation warning.
+
+Example:
+IF in buildpack API 0.3, `launch`, `cache`, and `build` key names change in the `layer.toml` and the previous keys are retained but deprecated in the 0.3 buildpack API spec:
+* IF a buildpack declares API 0.3 in it's `buildpack.toml`, AND uses the deprecated key name, THEN lifecycle will print a deprecation warning
+* IF a buildpack declares API 0.2 in it's `buildpack.toml`, AND uses the deprecated key name, THEN lifecycle will NOT print a deprecation warning, b/c the keys were not deprecated in the 0.2 version of the spec. 
+
+Eventually, we may wish to entirely deprecate older APIs. A future RFC should provide a strategy for deprecations and outline the upgrade path from 0.x to 1.0 APIs. This is purposefully omitted from this RFC so that we can make progress on new 0.x APIs while we work out the details.
+
+# How it Works
+[how-it-works]: #how-it-works
+## Spec
+The core team must not allow changes in a 0.x API that necessarily prevents buildpacks implementing different APIs from running together.
+The contractual build plan is a danger zone for such changes. This does not necessarily mean no changes can be made to the contractual build plan,
+but any such changes should be made in close collaboration with the implementation team to ensure that the lifecycle can manage compatibility.
+
+## Lifecycle
+The lifecycle will maintain code for correctly implementing all supported APIs and choose which API to implement according to the rules above.
+
+## Platforms
+Platforms that wish to check lifecycle compatibility may now treat all minor API versions as non-breaking with the exception
+of buildpack and platform `0.1` -> `0.2`. Platforms should always set the `CNB_PLATFORM_API` environment variable to indicate
+which version of the API they are using.
+
+## Buildpacks
+Buildpacks should continue to declare the minimum required buildpack API in their `buildpack.toml`.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The lifecycle must now deal with the complexity of implementing all 0.2+ APIs.
+
+The spec may not make certain types of changes to the contractual build plan before 1.0, even if those changes are desirable, because they make concurrent support for multiple APIs impossible.
+
+Buildpacks and platforms are likely to implement new APIs more slowly without a forcing function.
+
+If a buildpack or platform wishes to upgrade across several API versions, it may be difficult to easily discern the cumulative set of required changes from the spec alone.
```

Comment:
deprecations help mitigate this issue b/c they'll be in the API as you upgrade and marked clearly.